cmds/rm/rm.go

```diff
@@ -11,81 +11,142 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
 package main
 
 import (
 	"bufio"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
+	"syscall"
+	"strings"
+	"log"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
-	}
-	cmd = "rm [-Rrvi] file..."
-)
+// You can add more flags to this struct.
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
 
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
+// Helper method for interactive flag
+func iFlag(input *bufio.Scanner, printString string) bool {
```

Comment:
the name iFlag is a bit confusing given that this is a prompt of some sort, I'd consider a different name, just a thought.